memtier_benchmark.cpp

```diff
@@ -450,6 +454,20 @@ static int config_parse_args(int argc, char *argv[], struct benchmark_config *cf
                         return -1;
                     }
                     break;
+                case o_split:
+                    cfg->split = true;
+                    break;
+                case o_taskset:
+                    cfg->taskset = config_cpu_list(optarg);
+                    if (!cfg->taskset.is_defined()) {
+                        fprintf(stderr, "error: taskset must be expressed as [cpu1],...[cpuN] or [cpu1]-...[cpuN].\n");
+                        return -1;
+                    }
+                    if (!cfg->taskset.is_legal()) {
+                        fprintf(stderr, "error: an illegal cpu was passed in taskset list.\n");
+                        return -1;
+                    }
```

Comment:
not sure `is_define()` and `is_legal()` really needed. why not dealing invalid input with exceptions?
```cpp
try:
    cfg->taskset = config_cpu_list(optarg);
catch(std::Exception e):
    fprintf(stderr, "error: invalid input taskset must be expressed [cpu1],...[cpuN] or [cpu1]-...[cpuN].\n");
```

Reply:
is_define is needed. is_illegal was removed and parsing errors are now handled with exceptions